packages/realm-react/src/__tests__/useEmailPasswordAuth.test.tsx

```diff
@@ -0,0 +1,241 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+import React from "react";
+import { AppProvider } from "../AppProvider";
+import { waitFor, renderHook, act } from "@testing-library/react-native";
+
+import { AppConfigBuilder } from "@realm/app-importer";
+import { App, EmailPasswordAuth } from "realm";
+import { useEmailPasswordAuth } from "../useEmailPasswordAuth";
+import { OperationState } from "../types";
+import { baseUrl, importApp, testAuthOperation } from "./helpers";
+
+function renderEmailPasswordAuth(appId: string, baseUrl: string) {
+  const wrapper = ({ children }: { children: React.ReactNode }) => (
+    <AppProvider id={appId} baseUrl={baseUrl}>
+      {children}
+    </AppProvider>
+  );
+  return renderHook(() => useEmailPasswordAuth(), { wrapper });
+}
+
+describe("useEmailPassword", () => {
+  describe("with auto confirm", () => {
+    let appId: string;
+    beforeAll(async () => {
+      const config = new AppConfigBuilder("test-app");
+      config.authProvider({
+        name: "local-userpass",
+        type: "local-userpass",
+        config: {
+          autoConfirm: true,
+          resetPasswordUrl: "http://localhost/resetPassword",
+        },
+        disabled: false,
+      });
+      ({ appId } = await importApp(config.config));
+    });
+    it("can register and login with email/password, just by calling register", async () => {
+      const { result } = renderEmailPasswordAuth(appId, baseUrl);
+
+      await act(async () => {
+        result.current.register({ email: "test@test.com", password: "password" });
+        await waitFor(() => {
+          expect(result.current.result.pending).toEqual(true);
```

Comment:
I suggest that we test that `results.current.result.success` is false as it is an invariant that they cannot be true simultaneously. Likewise in the following assertions.

Reply:
As per Kræns suggestion, we won't test against pending, as this could be unstable and a race condition.  Rather we will wait for the expected ended result.